compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/DerivedConfigUtils.cpp

```diff
@@ -54,16 +54,30 @@ getThreadTileSizesFromLoopRanges(SmallVector<int64_t> loopRanges,
   }
 
   SmallVector<int64_t> tileSizes(loopRanges.size(), 0);
-  tileSizes.back() = vectorSize;
-  int64_t residualNumThreads = numThreads / (loopRanges.back() / vectorSize);
-  for (int i = tileSizes.size() - 2, e = 0; i >= e; --i) {
+  int64_t residualNumThreads = numThreads;
+  int i = tileSizes.size() - 1;
+  // Set as many inner tile sizes to 0 as possible to use the full vectorSize.
+  while (i >= 0) {
+    if (loopRanges[i] > vectorSize) {
+      tileSizes[i] = vectorSize;
+      residualNumThreads = numThreads / (loopRanges.back() / vectorSize);
+      i--;
```

Comment:
nit https://llvm.org/docs/CodingStandards.html#prefer-preincrement
```suggestion
      --i;
```